File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        result = 0
        l = r = 0
        while r < len(nums)-1:
            farthest = 0
            for i in range(l, r+1):
                farthest = max(farthest, i + nums[i])
            result += 1
            l, r = r+1, farthest
        return result

File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int res = 0;
        int l = 0;
        int r = 0;
        while (r < nums.length-1) {
            res++;
            int farthest = 0;
            for (int i = l; i < r+1; i++) {
                farthest = Math.max(farthest, i + nums[i]);
            }
            l = r+1;
            r = farthest;
        }       
        return res;
    }
}